### ISSUE TITLE
Ajax validation of semantically connected fields

### ISSUE THREAD

**Issue by tibee** [yiisoft/yii2#7383]
Hello!

Let's see this example. Having two fields in a product search form, `price_min` and `price_max` and AJAX validation. Any of these fields can be left blank.

If both are set, and `price_max` is smaller than `price_min`, both fields need to display validation error. 

But this error can be corrected by editing any of the two fields. The user edits i.e. `price_max`, enters a bigger value. Then, both fields validate OK, AJAX validation returns no errors, but the JS does not remove the error message from `price_min`, because it was not edited.

What is the workaround?
Thank you!

**Comment by samdark** [comment]
I guess you need your own validation class in this case. You can trigger validation using `validateAttribute(id)`.

**Comment by tibee** [comment]
It's OK for me, to write a custom validator class, adding errors to any fields using `$model->addError()`, or triggering the other field's validation, having `validateAttribute($model, $attribute)` to add the error on the field.

But I think, the JS side needs to be fixed somehow, because the full form validation on form submit and the AJAX validation on a single field event behave different.

1. If both fields have some validation error, my AJAX validation returns the following JSON:
```json
{"formname-price_min":["Price min error"],"formname-price_max":["Price max error"]}
```
If the AJAX request is triggered by editing the `price_min` field, only `price_min` field displays the error message. No matter that the JSON contains any other fields. The `price_max` stays 'green', no matter it has an error in the JSON.

2. If my AJAX validation returns:
```json
[]
```
Only the triggering fields deletes the displayed error message, all other fields keep it. As I wrote before.

**Comment by cebe** [comment]
@samdark I'm afraid the commit had the wrong issue number...

**Comment by samdark** [comment]
Won't be implemented in 2.0.
